pages/confirm.js

```diff
@@ -6,20 +6,17 @@ import { useRouter } from 'next/router'
 import LoadingSpinner from '../components/LoadingSpinner'
 import ErrorDisplay from '../components/ErrorDisplay'
 import api from '../lib/api-client'
+import useLoginRedirect from '../hooks/useLoginRedirect'
 
 const Confirm = () => {
   const [error, setError] = useState(null)
   const router = useRouter()
+  const { user, accessToken } = useLoginRedirect()
+  const [isLoading, setIsLoading] = useState(true)
```

Comment:
So it's considered best practice to not create additional state vars when the value could be derived from other state. Here isLoading could be derived from `emailToConfirm` and `error`.